### PR TITLE
Remove unnecessary datasource requests on clearing a search text (T847864)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 1]
```diff
@@ -567,37 +567,40 @@ QUnit.module('functionality', moduleSetup, () => {
...
     });
 
-    QUnit.test('Widget selects current value in the dropDownList if dxSelectBox with async data is opened on initialization (T822930)', function(assert) {
-        const selectBox = $('#selectBox').dxSelectBox({
-            deferRendering: true,
-            dataSource: {
-                load: () => {
-                    const d = $.Deferred();
+    [false, true].forEach((searchEnabled) => {
+        QUnit.test('Widget selects current value in the dropDownList if dxSelectBox with async data is opened on initialization (T822930)', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: Let's parameterize the test name subject.

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 2]
```diff
@@ -3344,6 +3399,9 @@ QUnit.module('Scrolling', {
 
         setTimeout(() => {
             assert.roughEqual(listInstance.scrollTop(), scrollingDistance, 150, 'scrollTop is correctly after new page load');
+            $('#qunit-fixture')
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: Les's do it in a separate PR?

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 3]
```diff
@@ -1848,6 +1851,58 @@ QUnit.module('editing', moduleSetup, () => {
...
+        const keyboard = keyboardMock($input);
+
+        $input.trigger('dxclick');
+        this.clock.tick(TIME_TO_WAIT);
+        keyboard.type('1');
+        this.clock.tick(TIME_TO_WAIT);
+        $('#qunit-fixture').trigger('dxpointerdown');
+        this.clock.tick(TIME_TO_WAIT);
+
+        assert.equal(loadMock.callCount, 2, 'load should not be called on init and on filter reset');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]:
```suggestion
        assert.equal(loadMock.callCount, 2, 'load should be called on init and on filter reset');
```

**Review by San4es — DISMISSED** [review]
LGTM

**Review by DokaRus — DISMISSED** [review]

**Review by San4es — APPROVED** [review]
